pkg/cliplugins/workspace/cmd/workspace.go

```diff
@@ -0,0 +1,570 @@
+	kubectlConfigOverrideFlags.AuthOverrideFlags.ImpersonateGroups.LongName = ""
+	kubectlConfigOverrideFlags.ContextOverrideFlags.AuthInfoName.LongName = ""
+	kubectlConfigOverrideFlags.ContextOverrideFlags.ClusterName.LongName = ""
+	kubectlConfigOverrideFlags.ContextOverrideFlags.Namespace.LongName = ""
+	kubectlConfigOverrideFlags.Timeout.LongName = ""
+
+	clientcmd.BindOverrideFlags(o.kubectlOverrides, cmd.Flags(), kubectlConfigOverrideFlags)
+
+	descriptionSuffix := " for workspace directory context"
+	workspaceDirectoryConfigOverrideFlags := clientcmd.RecommendedConfigOverrideFlags("workspace-directory-")
+
+	workspaceDirectoryConfigOverrideFlags.AuthOverrideFlags.ClientCertificate.LongName = ""
+	workspaceDirectoryConfigOverrideFlags.AuthOverrideFlags.ClientKey.LongName = ""
+	workspaceDirectoryConfigOverrideFlags.AuthOverrideFlags.Impersonate.LongName = ""
+	workspaceDirectoryConfigOverrideFlags.AuthOverrideFlags.ImpersonateGroups.LongName = ""
+	workspaceDirectoryConfigOverrideFlags.AuthOverrideFlags.Password.Description += descriptionSuffix
+	workspaceDirectoryConfigOverrideFlags.AuthOverrideFlags.Token.Description += descriptionSuffix
+	workspaceDirectoryConfigOverrideFlags.AuthOverrideFlags.Username.Description += descriptionSuffix
+
+	workspaceDirectoryConfigOverrideFlags.ContextOverrideFlags.AuthInfoName.LongName = ""
+	workspaceDirectoryConfigOverrideFlags.ContextOverrideFlags.ClusterName.LongName = ""
+	workspaceDirectoryConfigOverrideFlags.ContextOverrideFlags.Namespace.LongName = ""
+
+	workspaceDirectoryConfigOverrideFlags.ClusterOverrideFlags.APIVersion.LongName = ""
+	workspaceDirectoryConfigOverrideFlags.ClusterOverrideFlags.APIServer.Description += descriptionSuffix
+	workspaceDirectoryConfigOverrideFlags.ClusterOverrideFlags.APIServer.Default = "https://127.0.0.1:6444/services/applications/personal"
+	workspaceDirectoryConfigOverrideFlags.ClusterOverrideFlags.CertificateAuthority.Description += descriptionSuffix
+	workspaceDirectoryConfigOverrideFlags.ClusterOverrideFlags.InsecureSkipTLSVerify.Description += descriptionSuffix
+	workspaceDirectoryConfigOverrideFlags.ClusterOverrideFlags.TLSServerName.Description += descriptionSuffix
+
+	workspaceDirectoryConfigOverrideFlags.CurrentContext.Description += descriptionSuffix
+	workspaceDirectoryConfigOverrideFlags.CurrentContext.Default = "workspace-directory"
+	workspaceDirectoryConfigOverrideFlags.Timeout.LongName = ""
+
+	clientcmd.BindOverrideFlags(o.workspaceDirectoryOverrides, cmd.Flags(), workspaceDirectoryConfigOverrideFlags)
+	o.configAccess = clientcmd.NewDefaultClientConfigLoadingRules()
+
+	var err error
+	o.startingConfig, err = o.configAccess.GetStartingConfig()
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// Validate ensures that all required arguments and flag values are provided
+func (o *WorkspaceOptions) Validate() error {
+	return nil
+}
+
+func (o *WorkspaceOptions) ensureWorkspaceDirectoryContextExists() error {
```

Comment:
could this be functional without side-effects? Would be clearer. This modifies o heavily it seems.

Reply:
Nor really, since the idea is to add a context for the *workspace-directory* (== the `workspaces` virtual workspace) in the current Kubeconfig if none exists.
I commented the method to make it clear though.